mesmerize_viz/cnmfviewer.py

```diff
@@ -35,19 +34,66 @@ def __init__(
             contours: Union[Subplot, np.ndarray, Image],
             reconstructed: Union[Subplot, np.ndarray, Image],
             residuals: Union[Subplot, np.ndarray, Image],
-            #temporal: Union[Subplot, np.ndarray, Image],
+            # temporal: Union[Subplot, np.ndarray, Image],
             background: Union[Subplot, np.ndarray, Image],
             heatmap: Union[Subplot, np.ndarray, Image]
     ):
         self.input = input
         self.contours = contours
         self.reconstructed = reconstructed
         self.residuals = residuals
-        #self.temporal = temporal
+        # self.temporal = temporal
         self.background = background
         self.heatmap = heatmap
 
 
+class ContourSelection():
+    def __init__(
+            self,
+            gp: GridPlot,
+            coms,
+    ):
+        self.gp = gp
+        self.heatmap = self.gp.subplots[0, 1].scene.children[0]
+        self.image = None
+        self._contour_index = None
+
+        for child in self.gp.subplots[0, 0].scene.children:
+            if isinstance(child, pygfx.Image):
+                self.image = child
+                break;
+        if self.image is None:
+            raise ValueError("No image found!")
+        self.coms = np.array(coms)
+
+        self.image.add_event_handler(self.event_handler, "click")
+
+    # first need to add event handler for when contour is clicked on
+    # should also trigger highlighting in heatmap
+    def event_handler(self, event):
+        if self._contour_index is not None:
+            self.remove_highlight()
+            self.add_highlight(event)
+        else:
+            self.add_highlight(event)
+
+    def add_highlight(self, event):
+        click_location = np.array(event.pick_info["index"])
+        self._contour_index = np.linalg.norm((self.coms - click_location), axis=1).argsort()[0] + 1
+        line = self.gp.subplots[0, 0].scene.children[self._contour_index]
+        line.geometry.colors.data[:] = np.array([1.0, 1.0, 1.0, 1.0])
```

Comment:
once we implement a way to get graphics within a subplot conveniently, we can just use [`line_graphic.update_colors()`](https://github.com/kushalkolar/fastplotlib/blob/master/fastplotlib/graphics/line.py#L51)